Add jitterentropy as RNG (#4325)
randombit/botan: doc/api_ref/rng.rst

This RNG type allows using the RNG exported from a hardware token accessed via PKCS11.

Jitter_RNG
^^^^^^^^^^^^^^^^^

> This is a hardware RNG based on CPU timing jitter, using the
> `jitterentropy library <https://github.com/smuellerDD/jitterentropy-library>`_.

Reviewer: Nomenclature nit: Botan has a base class `Hardware_RNG` (that simply implements `clear() final { /* NOOP */ }`). But this is used for "actual" hardware RNG interfaces like RDRAND or TPM-based RNGs. It shouldn't be used as a base class for `Jitter_RNG` in my opinion. Though, to prevent confusion, perhaps:

```suggestion
This is an RNG based on low-level CPU timing jitter, using the
`jitterentropy library <https://github.com/smuellerDD/jitterentropy-library>`_.
```
